### PR TITLE
Collections Optional task

### PR DESCRIPTION
**Pull request by PavelYudzin**

### CODE REVIEW

File: java/src/main/java/com/company/automation/collections/optional_task/task01/Task01.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    1.   Ввести строки из файла, записать в список. Вывести строки в файл в обратном порядке.
+ */
+
+import java.io.*;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+public class Task01 {
+    private static final String pathToTask = "src\\main\\java\\com\\company\\automation\\collections\\optional_task\\Task01\\";
```
Comment (VitaliLebedzeu) [inline java/src/main/java/com/company/automation/collections/optional_task/task01/Task01.java]: use File.separator

Comment (VitaliLebedzeu) [inline java/src/main/java/com/company/automation/collections/optional_task/task01/Task01.java]: use File.separator

File: java/src/main/java/com/company/automation/collections/optional_task/task01/Task01.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        System.out.println(someLines);
+        List<String> revertListElements = new ArrayList<>(someLines);
+        Collections.reverse(revertListElements);
+        writeArrayListToFile(revertListElements);
+    }
+
+    private static List<String> getArrayListFromFile() {
+        List<String> lines = new ArrayList<>();
+        BufferedReader reader = null;
+        try {
```
Comment (VitaliLebedzeu) [inline java/src/main/java/com/company/automation/collections/optional_task/task01/Task01.java]: use try with resources...
try (BufferedReader reader = new BufferedReader(new FileReader(pathToTask + "Task01_Original.txt"))) {
...
} catch (...) {
...
}

File: java/src/main/java/com/company/automation/collections/optional_task/task01/Task01.java [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+                }
+            }
+        }
+        return lines;
+    }
+
+    private static void writeArrayListToFile(List<String> list) {
+        FileWriter writer = null;
+        try {
+            writer = new FileWriter(pathToTask + "Task01_Revert.txt");
```
Comment (VitaliLebedzeu) [inline java/src/main/java/com/company/automation/collections/optional_task/task01/Task01.java]: use try with resources

Comment (VitaliLebedzeu) [inline java/src/main/java/com/company/automation/collections/optional_task/task01/Task01.java]: please do not use the hardcode - "Task01_Revert.txt"

File: java/src/main/java/com/company/automation/collections/optional_task/task03/Task03.java [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+ */
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+
+public class Task03 {
+    private static final String PATH = new File("src" + File.separator + "main" + File.separator + "java" + File.separator + "com"
+            + File.separator + "company" + File.separator + "automation" + File.separator + "collections" + File.separator
+            + "optional_task" + File.separator + "task03" + File.separator + "directory").getAbsolutePath();
```
Comment (VitaliLebedzeu) [inline java/src/main/java/com/company/automation/collections/optional_task/task03/Task03.java]: separate string parameter to other  variable

File: java/src/main/java/com/company/automation/collections/optional_task/task03/Task03.java [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+            if (file.isDirectory()) {
+                directoryContent.add(file.getName());
+            }
+        }
+
+        for (File file : files) {
+            if (file.isFile()) {
+                directoryContent.add(file.getName());
+            }
+        }
```
Comment (VitaliLebedzeu) [inline java/src/main/java/com/company/automation/collections/optional_task/task03/Task03.java:33]: you can use LinkedList for escaping doble looping and use addFirst and addLast methods for sorting file list =)

**Review by VitaliLebedzeu — APPROVED** [review]

File: java/src/main/java/com/company/automation/collections/optional_task/task01/Task01.java (hunk 1 above)
Comment (PavelYudzin) [reply java/src/main/java/com/company/automation/collections/optional_task/task01/Task01.java]: fixed

Comment (PavelYudzin) [reply java/src/main/java/com/company/automation/collections/optional_task/task01/Task01.java]: fixed

File: java/src/main/java/com/company/automation/collections/optional_task/task01/Task01.java (hunk 2 above)
Comment (PavelYudzin) [reply java/src/main/java/com/company/automation/collections/optional_task/task01/Task01.java]: fixed

File: java/src/main/java/com/company/automation/collections/optional_task/task01/Task01.java (hunk 3 above)
Comment (PavelYudzin) [reply java/src/main/java/com/company/automation/collections/optional_task/task01/Task01.java]: fixed

Comment (PavelYudzin) [reply java/src/main/java/com/company/automation/collections/optional_task/task01/Task01.java]: fixed

File: java/src/main/java/com/company/automation/collections/optional_task/task03/Task03.java (hunk 4 above)
Comment (PavelYudzin) [reply java/src/main/java/com/company/automation/collections/optional_task/task03/Task03.java]: fixed

File: java/src/main/java/com/company/automation/collections/optional_task/task03/Task03.java (hunk 5 above)
Comment (PavelYudzin) [reply java/src/main/java/com/company/automation/collections/optional_task/task03/Task03.java:33]: fixed

**Pull request merged**